src/kOS/Execution/CPU.cs

```diff
@@ -1031,7 +1037,12 @@ private void ProcessTriggers()
                 }
             }
 
-            currentContext.InstructionPointer = currentInstructionPointer;
+            // since `run` opcodes don't work in triggers, we can use the opcode count to determine if the
+            // program has been aborted.  If the count isn't right, leave the pointer where it is.
```

Comment:
I guess for now we can accept it and flag it as a thing to test in the future after triggers stop being limited by IPU.